llvm/lib/Target/AMDGPU/Utils/AMDGPUPALMetadata.cpp

```diff
@@ -261,11 +261,8 @@ void AMDGPUPALMetadata::setEntryPoint(unsigned CC, StringRef Name) {
 
   // Set .entry_point which is defined
   // to be _amdgpu_<stage> and _amdgpu_cs for non-shader functions
-  SmallString<16> EPName("_amdgpu_");
-  raw_svector_ostream EPNameOS(EPName);
-  EPNameOS << getStageName(CC) + 1;
-  getHwStage(CC)[".entry_point"] =
-      MsgPackDoc.getNode(EPNameOS.str(), /*Copy=*/true);
+  getHwStage(CC)[".entry_point"] = MsgPackDoc.getNode(
+      (Twine("_amdgpu_") + getStageName(CC).drop_front()).str(), /*Copy=*/true);
```

Comment:
> TBH I'm not really sure if this makes the code better…

My original motivation was this case, where declaring a temporary string and a temporary ostream seems like overkill, when all we want to do is concatenate two strings.

Reply:
Yes, that would be fine, but for those with integers, etc., I don't think it makes the code better.

Reply:
FWIW, I feel like the new code is more readable.